src/hotspot/os_cpu/bsd_x86/os_bsd_x86.hpp

```diff
@@ -25,6 +25,10 @@
 #ifndef OS_CPU_BSD_X86_OS_BSD_X86_HPP
 #define OS_CPU_BSD_X86_OS_BSD_X86_HPP
 
+#if defined(COMPATIBLE_CDS_ALIGNMENT)
+#define CDS_CORE_REGION_ALIGNMENT (16*K)
+#endif
```

Comment:
Could you limit this to `_APPLE_`? I don't think this affects the BSDs.

(We really should separate BSD and MacOS sources at some point.)

Also maybe a comment like this: "Core region alignment is 16K to be able to run binaries built on MacOS x64 on MacOS aarch64." ?

Reply:
Fixed. Thanks.